addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+    },
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+    * @override
+    */
+    willStart: function () {
+        var url = '/web/webclient/locale/' + (document.documentElement.getAttribute('lang') || 'en_US').replace('-', '_');
+        var localeReady = ajax.loadJS(url);
+        return Promise.all([this._super.apply(this, arguments), localeReady]);
+    },
+
+    /**
+    * @override
+    */
+    start: function () {
+        var self = this;
+        self.surveyToken = self.$target.data('surveyToken');
+        self.answerToken = self.$target.data('answerToken');
+
+        var $timer = $('.o_survey_timer');
+        if ($timer.length) {
+            self.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(null, $timer.data());
+            self.surveyTimerWidget.attachTo($timer);
+            self.surveyTimerWidget.on('time_up', self, function (ev) {
+                self.$target.find('button[name="button_submit"]').click();
+            });
+        }
+
+        _.each(this.$('div.o_survey_form_date'), function (date_group) {
+            self._updateDateForDisplay($(date_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_radio'), function (radio_group) {
+            self._updateSimpleRadioForDisplay($(radio_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_select'), function (select_group) {
+            self._updateSimpleSelectForDisplay($(select_group));
+        });
+
+        return this._super.apply(this, arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    // EVENTS
+    // -------------------------------------------------------------------------
+    _onChangeRadio: function (event) {
+        var $radio_group = $(event.currentTarget).parents('.o_survey_form_simple_radio');
+        return this._updateSimpleRadioForDisplay($radio_group);
+    },
+
+    _onChangeSelect: function (event) {
+        var $select_group = $(event.currentTarget).parents('.o_survey_form_simple_select');
+        return this._updateSimpleSelectForDisplay($select_group);
+    },
+
+    _onChangeMultipleCheckbox: function (event) {
+        var $elem = $(event.currentTarget);
+        var $comment_input = $elem.parents('.o_survey_form_multiple').find('input[type="text"]');
+        if ($elem.prop('checked')) {
+            $comment_input.focus();
+        } else {
+            $comment_input.val('');
+        }
+    },
+
+    _onSubmit: function (event) {
+        var self = this;
+        event.preventDefault();
+        var $elem = $(event.currentTarget);
+        var submitValue = $elem.val();
+        var $form = $elem.parents('form.o_survey_form');
+        var formData = $form.serializeArray();
```

Comment:
Maybe use standard JS stuff for this

Have a look at 
- `var formData = new FormData($form[0]);`
- https://stackoverflow.com/questions/41431322/how-to-convert-formdatahtml5-object-to-json